### PR TITLE
Add batch timeout override

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Add environment variable option to set batch timeout limit.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py [hunk 1]
```diff
@@ -23,16 +23,16 @@ def __init__(self, run, input_data, evaluator_name=None, aggregated_metrics=None
...
-    def get_result_df(self, exclude_inputs=False):
-        result_df = self.run.result(timeout=BATCH_RUN_TIMEOUT)
+    def get_result_df(self, exclude_inputs=False, batch_run_timeout=BATCH_RUN_TIMEOUT):
+        result_df = self.run.result(timeout=batch_run_timeout)
         if exclude_inputs:
             result_df = result_df.drop(columns=[col for col in result_df.columns if col.startswith("inputs.")])
         return result_df
 
-    def get_aggregated_metrics(self):
+    def get_aggregated_metrics(self, batch_run_timeout=BATCH_RUN_TIMEOUT):
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py:32]: Can users control the timeout ?

Comment (diondrapeck) [reply src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py]: yes, this was just setting the BATCH_RUN_TIMEOUT constant as a default; however, I've now changed this to be read via environment variable rather than a method argument.

File: src/promptflow-evals/promptflow/evals/_constants.py [hunk 2]
```diff
@@ -23,4 +23,5 @@ class Prefixes:
 
 CONTENT_SAFETY_DEFECT_RATE_THRESHOLD_DEFAULT = 4
 
-BATCH_RUN_TIMEOUT = 3600
+BATCH_RUN_TIMEOUT_DEFAULT = 3600
+BATCH_RUN_TIMEOUT_ENV_VAR = "BATCH_RUN_TIMEOUT"
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/_constants.py:27]: May prefix with 'PF', like: 
PF_BATCH_RUN_TIMEOUT

Comment (wangchao1230) [reply src/promptflow-evals/promptflow/evals/_constants.py:27]: Actually there seems to already have an env called this.
PF_BATCH_TIMEOUT_SEC

Comment (diondrapeck) [reply src/promptflow-evals/promptflow/evals/_constants.py]: updated

File: src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py [hunk 3]
```diff
@@ -24,15 +25,17 @@ def __init__(self, run, input_data, evaluator_name=None, aggregated_metrics=None
         self.aggregated_metrics = aggregated_metrics
 
     def get_result_df(self, exclude_inputs=False):
-        result_df = self.run.result(timeout=BATCH_RUN_TIMEOUT)
+        batch_run_timeout = get_int_env_var(PF_BATCH_TIMEOUT_SEC, PF_BATCH_TIMEOUT_DEFAULT)
+        result_df = self.run.result(timeout=batch_run_timeout)
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py:29]: why do we need to set the timeout here? As long we set the timeout on the execution part, I think we can omit the setting here?

Comment (wangchao1230) [reply src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/code_client.py:29]: @thy09 to make some comments

File: src/promptflow-devkit/promptflow/batch/_batch_engine.py [hunk 4]
```diff
@@ -49,6 +49,7 @@
 from promptflow.contracts.flow import Flow
 from promptflow.contracts.run_info import FlowRunInfo, Status
 from promptflow.contracts.types import AttrDict
+from promptflow.evals._constants import PF_BATCH_TIMEOUT_SEC
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/batch/_batch_engine.py:52]: do we need to add any changelog for this PR?

Comment (PeiwenGaoMS) [inline src/promptflow-devkit/promptflow/batch/_batch_engine.py:52]: `promptflow-devkit` should not depend on `promptflow-evals`. Please revert the changes in `_batch_engine.py`.

Comment (diondrapeck) [reply src/promptflow-devkit/promptflow/batch/_batch_engine.py:52]: Will do.
